contracts/ReaperAutoCompoundScreamLeverage.sol

```diff
@@ -771,4 +774,12 @@ contract ReaperAutoCompoundScreamLeverage is ReaperBaseStrategy {
         IERC20Upgradeable(WFTM).safeDecreaseAllowance(UNI_ROUTER, IERC20Upgradeable(WFTM).allowance(address(this), UNI_ROUTER));
         IERC20Upgradeable(SCREAM).safeDecreaseAllowance(UNI_ROUTER, IERC20Upgradeable(SCREAM).allowance(address(this), UNI_ROUTER));
     }
+
+    /**
+     * @dev Helper modifier for functions that need to update the internal balance at the end of their execution.
+     */
+    modifier doUpdateBalance {
+        _;
+        updateBalance();
```

Comment:
Modifier adds a lot of contract size doesn't it? Since it just pastes in the whole function. I think it is preferable to call the updateBalance() where needed to save contract size and keep some of these other functions.